### ISSUE TITLE
resource "newrelic_synthetics_monitor" returns entity GUID instead of monitor ID

### ISSUE THREAD

**Issue by reschenburgIDBS** [newrelic/terraform-provider-newrelic#1981]
- [x] Your New Relic `provider` [configuration](#terraform-configuration-files) (sensitive details redacted)
- [x] A list of [affected resources](#affected-resources) and/or data sources
- [x] The [configuration](#terraform-configuration-files) of the resources and/or data sources related to the bug report (i.e. from the list mentioned above)
- [x] Description of the [current behavior](#actual-behavior) (the bug)
- [x] Description of the [expected behavior](#expected-behavior)
- [x] Any related [log output](#debug-output)

### Terraform Version
`1.0.1` - although the issue is very much unrelated to the terraform version

### Affected Resource(s)
- `newrelic_synthetics_monitor` 

### Terraform Configuration

```hcl

required_providers {
  newrelic = {
    source  = "newrelic/newrelic"
    version = ">= 3.0"
  }
}

provider "newrelic" {
  account_id = "xxx"
  api_key    = module.ssm_newrelic_key.parameters.synthetics_api.value
  region     = "US"
}

resource "newrelic_synthetics_monitor" "tcp_ping" {
  count      = length(local.service_endpoints)
  name       = local.service_endpoints[count.index].name
  type       = "SIMPLE"
  period     = lookup(local.frequency_to_period, var.monitor_frequency, var.monitor_frequency) 
  status     = var.status
  locations_public  = local.locations
  uri        = local.service_endpoints[count.index].url
  verify_ssl = true
  lifecycle {
    ignore_changes = [status]
  }
}

output "newrelic_monitor_id" {
  value = length(newrelic_synthetics_monitor.tcp_ping) > 0 ? newrelic_synthetics_monitor.tcp_ping[0].id : ""
}

```

### Actual Behavior
Since the 3.0 provider upgrade, the `newrelic_synthetics_monitor.tcp_ping[0].id` returns the `Entity guid` of the Monitor.
Any API calls to this entity guid return a 404 (as the API expects the monitor id instead).

### Expected Behavior
Previously, the ID exposed by the synthetics monitor resource was the actual `Monitor ID` 

As a result, I was able to use the terraform output of this ID to inform an API call to enable the monitor.

My workflow is/was as follows: deploy infrastructure with terraform, deploy microservice into K8s, use API call to enable the NR monitor on the deployed microservice - this no longer works, as the ID can't be used directly against the API anymore. 

### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:
1. deploy synthetic monitor, using terraform.
2. curl the synthetics API with the returned `id` (`GET` `https://synthetics.newrelic.com/synthetics/api/v3/monitors/<ID>`)
3. -> receive 404

### Important Factoids
- I'm storing the monitor ID in parameter store, so I can see the ID changing after the provider upgrade
- this is not mentioned in the v2 -> v3 upgrade documentation
- I'd be more than happy for the resource to return _both_ IDs - but for backwards compatibility I'd prefer the `id` to be the monitor id as it used to be and to add a new parameter for `guid` instead.

**Comment by reschenburgIDBS** [comment]
Having looked at the actual code changes around the ID/GUID, I'm using a hack at the moment to get the monitor ID:
(if, like me, you are doing this in Jenkins):
``` groovy
monitorID = new String(nrGuid.decodeBase64()).split("\\|")[3]
```

**Comment by reschenburgIDBS** [comment]
for those wondering.. the following code snippet doesn't work in terraform v1.0.1 - v1.1.9:

``` hcl
monitor_id_new = split("|", base64decode(newrelic_synthetics_monitor.tcp_ping.id))[3]
```

with
```
│ Error: Error in function call
│ 
│   on <console-input> line 1:
│   (source code not available)
│ 
│ Call to function "base64decode" failed: failed to decode base64 data "<monitor GUID>".
```

**Comment by kidk** [comment]
Hi @reschenburgIDBS 

This works as expected because of the switch from our old REST API's which were using the monitor ID to the GraphQL API's which use GUID by default. 

We're going to look to see if we can add the monitor_id as an attribute to the resource, but in the meantime the workaround by decoding the guid with base64 should keep working.

I'd recommend also exploring the new GraphQL API as there are plans in the works to deprecate the REST based Synthetics API's. As an example you can get information about a Synthetics check with [this GraphQL query](https://api.newrelic.com/graphiql?#query=%7B%0A%20%20actor%20%7B%0A%20%20%20%20entity%28guid%3A%20%22%22%29%20%7B%0A%20%20%20%20%20%20...%20on%20Entity%20%7B%0A%20%20%20%20%20%20%20%20guid%0A%20%20%20%20%20%20%20%20name%0A%20%20%20%20%20%20%20%20tags%20%7B%0A%20%20%20%20%20%20%20%20%20%20values%0A%20%20%20%20%20%20%20%20%20%20key%0A%20%20%20%20%20%20%20%20%7D%0A%20%20%20%20%20%20%7D%0A%20%20%20%20%7D%0A%20%20%7D%0A%7D%0A). 

> this is not mentioned in the v2 -> v3 upgrade documentation
We will modify the guide to include this.

I hope this helps.

**Comment by reschenburgIDBS** [comment]
Thanks for pointer to the GraphQL API @kidk, here I've managed to enable/disable my monitor based on the GUID!

Any ideas why terraform doesn't like to base64decode the GUID? I'm maintaining an internal module to create monitors, which would be the most central place for me to do the decoding at the moment, before I can update all of our API calls to use GraphQL. 
Might be a question for the terraform folks instead as it doesn't look like the guid is using any kind of special characters or encoding and every other method I've tried is able to decode it just fine.. 🤷‍♂️
